drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -153,37 +274,174 @@ static void esp_wifi_event_task(void)
 	}
 }
 
-static void eth_esp32_init(struct net_if *iface)
+static int esp32_wifi_disconnect(const struct device *dev)
+{
+	struct esp32_wifi_runtime *data = dev->data;
+	int ret = esp_wifi_disconnect();
+
+	if (ret != ESP_OK) {
+		LOG_INF("Failed to disconnect from hotspot");
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+static int esp32_wifi_connect(const struct device *dev,
+			    struct wifi_connect_req_params *params)
+{
+	struct esp32_wifi_runtime *data = dev->data;
+	int ret;
+
+	if (data->state == ESP32_STA_CONNECTING || data->state == ESP32_STA_CONNECTED) {
+		wifi_mgmt_raise_connect_result_event(esp32_wifi_iface, -1);
+		return -EALREADY;
+	}
+
+	if (data->state != ESP32_STA_STARTED) {
+		LOG_ERR("Wi-Fi not in station mode");
+		wifi_mgmt_raise_connect_result_event(esp32_wifi_iface, -1);
+		return -EIO;
+	}
+
+	data->state = ESP32_STA_CONNECTING;
+
+	wifi_config_t wifi_config;
+
+	memset(&wifi_config, 0, sizeof(wifi_config_t));
+
+	memcpy(wifi_config.sta.ssid, params->ssid, params->ssid_length);
+	wifi_config.sta.ssid[params->ssid_length] = '\0';
+
+	if (params->security == WIFI_SECURITY_TYPE_PSK) {
+		memcpy(wifi_config.sta.password, params->psk, params->psk_length);
+		wifi_config.sta.password[params->psk_length] = '\0';
+		wifi_config.sta.threshold.authmode = WIFI_AUTH_WPA2_PSK;
+	} else {
+		wifi_config.sta.threshold.authmode = WIFI_AUTH_OPEN;
```

Comment:
with latest changes that extend the params->security, this is not true, so, need to add an explicit security check.

Reply:
@krish2718, that's true. I'll add those new securities in the loop.